asserts/model.go

```diff
@@ -707,6 +738,20 @@ func assembleModel(assert assertionBase) (Assertion, error) {
 			grade = ModelGrade(gradeStr)
 		}
 
+		storageSafetyStr, err := checkOptionalString(assert.headers, "storage-safety")
+		if err != nil {
+			return nil, err
+		}
+		if storageSafetyStr != "" && !strutil.ListContains(validStorageSafeties, storageSafetyStr) {
+			return nil, fmt.Errorf("storage-safety for model must be %s, not %q", strings.Join(validStorageSafeties, "|"), storageSafetyStr)
+		}
+		if storageSafetyStr != "" {
+			storageSafety = ModelStorageSafety(storageSafetyStr)
+		}
+		if grade == ModelSecured && storageSafety == ModelStorageSafetyPreferUnencrypted {
+			return nil, fmt.Errorf(`"grade: secured" cannot have "storage-safety: prefer-unencrypted"`)
```

Comment:
should we actually be more strict and require that `grade: secured` only have `ModelStorageSafetyEncrypted`, aiui the "prefer-encrypted" option would still (eventually) allow a user to tell the system to install unencrypted with a `grade: secured` model which seems unintentional to me

Reply:
yes, correct.  For secured storage-safety  should either be not specified or redundantly specified as "encrypted".